File: Easy/tassaistanzaconf/meta_tassaistanzaconfdefaultview/meta_tassaistanzaconfdefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_tassaistanzaconfdefaultview
{
    public class Meta_tassaistanzaconfdefaultview : Meta_easydata 
	{
        public Meta_tassaistanzaconfdefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "tassaistanzaconfdefaultview") {
				Name = "Definizione dei costi delle istanze";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idtassaistanzaconf"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "tassaistanzaconf_aa", "Anno accademico", nPos++);
						DescribeAColumn(T, "costoscontodef_title", "Costo", nPos++);
						DescribeAColumn(T, "istanzakind_title", "Tipo di istanza", nPos++);
						DescribeAColumn(T, "tassaistanzaconf_nullaosta", "Nullaosta", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/tassaistanzaconf/meta_tassaistanzaconfdefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_tassaistanzaconfdefaultview() {
        MetaData.apply(this, ["tassaistanzaconfdefaultview"]);
        this.name = 'meta_tassaistanzaconfdefaultview';
    }

    meta_tassaistanzaconfdefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_tassaistanzaconfdefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'aa', 'Anno accademico', null, 10, 9);
						this.describeAColumn(table, 'costoscontodef_title', 'Costo', null, 30, 2024);
						this.describeAColumn(table, 'istanzakind_title', 'Tipo di istanza', null, 40, 50);
						this.describeAColumn(table, 'tassaistanzaconf_nullaosta', 'Nullaosta', null, 50, null);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idtassaistanzaconf"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			//$getSorting$

        });

    window.appMeta.addMeta('tassaistanzaconfdefaultview', new meta_tassaistanzaconfdefaultview('tassaistanzaconfdefaultview'));

	}());
